### ISSUE TITLE
Unable to run build file

### ISSUE THREAD

**Issue by shruti1411** [dotnet/runtimelab#1166]
I was trying to run source code by using build.sh file. But I'm unable to build CoreCLR Component and shows following error.

 
<img width="869" alt="RuntimeCoreCLR" src="https://user-images.githubusercontent.com/81982171/119509078-ed0ca780-bd8d-11eb-945b-32de45e96a8b.png">

Hoping for your positive consideration for this.

**Comment by jkotas** [comment]
What is the exact build command line that you are using?

The instruction for building this branch are here: https://github.com/dotnet/runtimelab/blob/feature/NativeAOT/docs/workflow/building/coreclr/nativeaot.md#building

**Comment by shruti1411** [comment]
> What is the exact build command line that you are using?
> 
> The instruction for building this branch are here: https://github.com/dotnet/runtimelab/blob/feature/NativeAOT/docs/workflow/building/coreclr/nativeaot.md#building

I tried this above link and run following command

./build.sh nativeaot+libs+nativeaot.packages -rc Debug -lc Release 

but still get same error.


<img width="872" alt="CLRfailed" src="https://user-images.githubusercontent.com/81982171/119604908-7e6f2e80-be0d-11eb-8b88-5f14c192f826.png">


<img width="882" alt="RUNCommand" src="https://user-images.githubusercontent.com/81982171/119606369-1a9a3500-be10-11eb-974b-ae1e488662aa.png">

**Comment by jkotas** [comment]
Could you please delete you `artifacts` directory and try to build again?

If it still fails, could you please share the whole build log printed on the console?

**Comment by shruti1411** [comment]
Hii sir,
I checked with deleting artifacts folder but still get same error. I'm
attaching log file from console.Please check it once.

Thanks and Regards,
Shruti Patil.

**Comment by jkotas** [comment]
Looks like attaching the log via email did not work. I am not able to see it. Could you please attach it via browser?

**Comment by shruti1411** [comment]
Please find attached doc. of log file from console.

[BuildLog.docx](https://github.com/dotnet/runtimelab/files/6552271/BuildLog.docx)

**Comment by jkotas** [comment]
This is the problem:

    !!! Cannot find libssl and System.Security.Cryptography.Native cannot build
    without it.  Try installing libssl-dev (on Linux, but this may vary by
    distro) or openssl (on macOS) !!!.  See the requirements document for your
    specific operating system:
    https://github.com/dotnet/runtime/tree/main/docs/workflow/requirements.

**Comment by shruti1411** [comment]
By installing all requirement mentioned in above URL still get same error

**Comment by shruti1411** [comment]
how to install System.Security.Cryptography.Native ???

I installed the whole command for linux mentioned in above URL. But still get same error.

**Comment by jkotas** [comment]
System.Security.Cryptography.Native is built in this repo. You do not need to install  System.Security.Cryptography.Native . But you need to install what it depends on.

What is the Linux distro that you are running on?

**Comment by shruti1411** [comment]
I am running on Ubuntu 18.04

**Comment by jkotas** [comment]
Does this report any errors (it is the command from https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/linux-requirements.md#toolchain-setup):
```
sudo apt-get install -y cmake llvm-9 clang-9 \
build-essential python curl git lldb-6.0 liblldb-6.0-dev \
libunwind8 libunwind8-dev gettext libicu-dev liblttng-ust-dev \
libssl-dev libnuma-dev libkrb5-dev zlib1g-dev ninja-build
```

**Comment by shruti1411** [comment]
Yes....I run this command for Ubuntu 18.04....and still get same error like
previous.

**Comment by jkotas** [comment]
It looks like that cmake or OpenSSL end up misconfigured on your system somehow.

Could you please run this small test - to double check that missing OpenSSL is really the problem:

1. Create a new scratch directory, e.g.:
```
mkdir ~/test
cd ~/test
```

2. In that directory, create `CMakeLists.txt` that contains:
```
project(Project)

cmake_minimum_required(VERSION 3.14.5)

find_package(OpenSSL)

if(NOT OPENSSL_FOUND)
   message(FATAL_ERROR "!!! Cannot find OpenSSL")
endif(NOT OPENSSL_FOUND)

if(OPENSSL_FOUND)
   message("OpenSSL found!")
endif(OPENSSL_FOUND)
```

3. Run cmake:
```
cmake . 
```

Does this small test still fail with `Cannot find OpenSSL` error?

**Comment by shruti1411** [comment]
Haven't received such type of error. In CMakeLists.txt use 3.10.2 version instead of 3.14.5.  

<img width="885" alt="CMakeOPENSSL" src="https://user-images.githubusercontent.com/81982171/119953265-7228d400-bfbb-11eb-93a0-0ed83ed54e38.png">

**Comment by jkotas** [comment]
>  3.10.2

This cmake version may be a problem. The repo requires 3.14.5 according to https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/linux-requirements.md#toolchain-setup

**Comment by shruti1411** [comment]
run on updated version of cmake ....but still can't happened anything else.....gives same error like previous.

**Comment by jkotas** [comment]
I am running out of ideas why it is failing for you. The next thing I would try is to run the cmake with `--trace` and try to figure out what went wrong from the trace, compare it with the trace for the standalone test where it works fine.

**Comment by ShreyasJejurkar** [comment]
@shruti1411 After updating cmake check version with `cmake --version` does it show `3.14.5` or `3.10.2`? It should show `3.14.5`!
Back in the day, cmake wouldn't pick up a new version unless I restart the machine or restart the shell again. So make sure you give try to that as well!

**Comment by shruti1411** [comment]
> @shruti1411 After updating cmake check version with `cmake --version` does it show `3.14.5` or `3.10.2`? It should show `3.14.5`!
> Back in the day, cmake wouldn't pick up a new version unless I restart the machine or restart the shell again. So make sure you give try to that as well!


I updated newer version of cmake which 3.20.3 and run build command but it shows same error .....I am not able to install cmake 3.14.5 it shows cmake 3.14.5 was not found.


<img width="502" alt="Cmake3 14 5" src="https://user-images.githubusercontent.com/81982171/120172637-e2845f00-c220-11eb-8ca3-3533abc83a4e.png">

**Comment by shruti1411** [comment]
> I am running out of ideas why it is failing for you. The next thing I would try is to run the cmake with `--trace` and try to figure out what went wrong from the trace, compare it with the trace for the standalone test where it works fine.


I run cmake with --trace command for standalone test............

Attaching log file where it works fine with --trace as well cmake  . command for reference.
[cmake.txt](https://github.com/dotnet/runtimelab/files/6569376/cmake.txt)

The following log file works on my system..... where it shows difference in log from 
[cmaketrace.docx](https://github.com/dotnet/runtimelab/files/6569773/cmaketrace.docx)
above file.

**Comment by jkotas** [comment]
The logs show that running cmake from `/usr/share/cmake-3.20/` works and running cmake via snap from `/snap/cmake/876/share/cmake-3.20/` does not work. Looks like snap virtualization is hidding openssl libraries from cmake.

Any chance you can install all tools required by the build natively without snap? There is probably a way to make the build work with snap installed tools too, but it would require more investigation.

**Comment by Nathair** [comment]
> 
> 
> I was trying to run source code by using build.sh file. But I'm unable to build CoreCLR Component and shows following error.
> 
> <img alt="RuntimeCoreCLR" width="869" src="https://user-images.githubusercontent.com/81982171/119509078-ed0ca780-bd8d-11eb-945b-32de45e96a8b.png">
> 
> Hoping for your positive consideration for this.

i have same problem. ninja cant find config file build.ninja. Jsut creete them in src\coreclr. Run build.cmd clean and build.cmd nativeaot+libs+nativeaot.packages  -rc Debug -lc Release

**Comment by MichalStrehovsky** [comment]
Closing stale issues.
